go/vt/schemadiff/table.go

```diff
@@ -1140,13 +1140,13 @@ func (c *CreateTableEntity) diffKeys(alterTable *sqlparser.AlterTable,
 		t2KeysMap[key.Info.Name.String()] = key
 	}
 
-	dropKeyStatement := func(name sqlparser.IdentifierCI) *sqlparser.DropKey {
+	dropKeyStatement := func(info *sqlparser.IndexInfo) *sqlparser.DropKey {
 		dropKey := &sqlparser.DropKey{}
-		if strings.EqualFold(dropKey.Name.String(), "PRIMARY") {
+		if strings.EqualFold(info.Type, sqlparser.PrimaryKeyTypeStr) {
```

Comment:
This wasn't even checking anything at all. It checked the name of `dropKey` but that was just initialized to a new struct so this check would never be true at all. By passing in the more details from `info` we can check the actual type.